src/data/LongData.java

```diff
@@ -0,0 +1,42 @@
+package data;
+
+import errors.CommandException;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LongData extends Data<Long> {
+
+    public LongData(File input) throws CommandException {
+        super(input);
+    }
+
+    public void mapInvalid(List<String> data) {
+        data
+                .stream()
+                .flatMap(s -> Arrays.stream(s.split(" ")))
+                .filter(x -> x.matches(".*\\D.*"))
+                .map(s -> String.format("\"%s\" is not a long. It will be skipped.\n", s))
+                .forEach(System.out::print);
+    }
+
+    @Override
+    public List<Long> mapData(List<String> data) {
```

Comment:
Currently mapInvalid and mapData duplicate logic. We can do it in one method.
Also current implementation doesn't filter out huge values. E.g. the following example is not a valid long value:
111111111111111111111111

```suggestion
  @Override
    public List<Long> mapData(List<String> data) {
        return data
                .stream()
                .map(String::trim)
                .flatMap(s -> Arrays.stream(s.split(" ")))
                .map(x -> {
                    try {
                        return Long.parseLong(x);
                    } catch (NumberFormatException e) {
                        System.out.printf("\"%s\" is not a long. It will be skipped.\n", x);
                        return null;
                    }
                })
                .filter(Objects::nonNull)
                .sorted(Comparator.naturalOrder())
                .collect(Collectors.toList());
    }
```